File: client/src/app/models/basket.ts
import { v4 as uuidv4 } from 'uuid';
import { IProduct } from './product';

export interface IBasket {
  id: string;
  basketItems: IBasketItem[];
  clientSecret?: string;
  paymentIntentId?: string;
  deliveryMethodId?: number;
  shippingPrice?: number;
}



export interface IBasketItem {
  id: number;
  name: string;
  description: string;
  price: number;
  imgUrl: string;
  productCategoryName: string;
  quantity: number
}


export class Basket implements IBasket {
  basketItems: IBasketItem[] = [];
  clientSecret?: string;
  paymentIntentId?: string;
  deliveryMethodId?: number;
  shippingPrice?: number;
  id: string = uuidv4();
}

export interface IBasketTotals {
  shipping: number;
  subtotal: number;
  total: number;
}

File: backendApi/BackendApi/Core/Models/Entities/Basket.cs
using System.Collections.Generic;

namespace BackendApi.Core.Models.Entities
{
    public class Basket
    {
        public Basket()
        {
        }

        public Basket(string id)
        {
            this.Id = id;
        }

        public List<BasketItem> BasketItems {get;set;} = new List<BasketItem>();

        public string Id { get; set; }
        public int? DeliveryMethodId { get; set; }
        public string ClientSecret { get; set; }
        public decimal ShippingPrice { get; set; }
        public string PaymentIntentId { get; set; }
    }
}